File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetstransferpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsTransferPagedResponseHrefModel } from "./codatdatacontractsdatasetstransferpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsTransferPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsTransferPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsTransferPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsTransferPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsTransferPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsTransferPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsTransferPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsTransferPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsTransferPagedResponseHrefModel;
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_transferpagedresponselinksmodel.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import codat_datacontracts_datasets_transferpagedresponsehrefmodel as shared_codat_datacontracts_datasets_transferpagedresponsehrefmodel


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsTransferPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_transferpagedresponsehrefmodel.CodatDataContractsDatasetsTransferPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current') }})
    next: Optional[shared_codat_datacontracts_datasets_transferpagedresponsehrefmodel.CodatDataContractsDatasetsTransferPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[shared_codat_datacontracts_datasets_transferpagedresponsehrefmodel.CodatDataContractsDatasetsTransferPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    self: Optional[shared_codat_datacontracts_datasets_transferpagedresponsehrefmodel.CodatDataContractsDatasetsTransferPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self') }})
    
